provider/provider.go

```diff
@@ -10,6 +10,37 @@ import (
 	"github.com/go-spatial/tegola/internal/log"
 )
 
+type providerType uint8
+
+const (
+	TypeStd providerType = iota
+	TypeMvt
+)
+
+func (pt providerType) Prefix() string {
+	if pt == TypeMvt {
+		return "mvt_"
+	}
+	return ""
+}
+
+func (pt providerType) String() string {
+	if pt == TypeMvt {
+		return "MVT Provider"
+	}
+	return "Standard Provider"
+}
+
+type providerFilter uint8
+
+const (
+	ProviderFilterNone providerFilter = 0
+	ProviderFilterStd                 = 1 << iota
+	ProviderFilterMVT
+
+	ProviderFilterAll = ProviderFilterStd | ProviderFilterMVT
+)
```

Comment:
There's [another issue](https://github.com/go-spatial/tegola/issues/610) around filtering features at the provider layer. I'm thinking we might run into a naming collision with this filter concept and the data filter concept. Maybe we make these const private if possible, or be prepared to potentially change the name for data filtering.